src/pip/_internal/index.py

```diff
@@ -59,8 +59,40 @@
 logger = logging.getLogger(__name__)
 
 
-def _get_content_type(url, session):
-    """Get the Content-Type of the given url, using a HEAD request"""
+def _match_vcs_scheme(url):
+    """Look for VCS schemes in the URL.
+
+    Returns the matched VCS scheme, or None if there's no match.
+    """
+    from pip._internal.vcs import VcsSupport
+    for scheme in VcsSupport.schemes:
+        if url.lower().startswith(scheme) and url[len(scheme)] in '+:':
+            return scheme
+    return None
+
+
+def _is_filename_like_archive(filename):
+    """Check whether the filename looks like an archive.
+    """
+    for bad_ext in ARCHIVE_EXTENSIONS:
+        if filename.endswith(bad_ext):
+            return True
+    return False
+
+
+class _NotHTML(Exception):
+    pass
+
+
+def _ensure_html_header(headers):
```

Comment:
Especially since you're relying on exceptions for flow control, I would document in the docstring that this can raise `_NotHTML`.